distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP.enso

```diff
@@ -235,66 +204,95 @@ resolve_headers req =
         all_headers + default_content_type
 
 ## PRIVATE
-   Generate body publisher and optional form content boundary
-resolve_body_to_publisher_and_boundary : Request_Body -> Pair BodyPublisher Text
-resolve_body_to_publisher_and_boundary body:Request_Body =
+   Generate body publisher, optional form content boundary and optional hash from the body
+_resolve_body : Request_Body -> Function | Nothing -> Pair BodyPublisher Text
+private _resolve_body body:Request_Body hash_function =
     body_publishers = HttpRequest.BodyPublishers
     case body of
         Request_Body.Text text encoding _ ->
             body_publisher = case encoding of
                 Nothing -> body_publishers.ofString text
                 _ : Encoding ->  body_publishers.ofString text encoding.to_java_charset
-            Pair.new body_publisher Nothing
+            hash = if hash_function.is_nothing then "" else hash_function (text.bytes (encoding.if_nothing Encoding.utf_8))
+            [body_publisher, Nothing, hash]
         Request_Body.Json x ->
             json = x.to_json
-            json.if_not_error <|
-                Pair.new (body_publishers.ofString json) Nothing
+            hash = if hash_function.is_nothing then "" else hash_function json.bytes
+            json.if_not_error <| [body_publishers.ofString json, Nothing, hash]
         Request_Body.Binary file ->
             path = File_Utils.toPath file.path
-            Pair.new (body_publishers.ofFile path) Nothing
+            ## ToDo: Support hashing a file.
+            [body_publishers.ofFile path, Nothing, Nothing]
```

Comment:
What about this ToDo?

Should we throw an error if a hash was requested?

What will happen if I try to make an AWS signed request with a file payload?